beluga_ros/test/test_point_cloud.cpp

```diff
@@ -0,0 +1,887 @@
+
+    // Fill the PointCloud2 message
+    typename std::vector<T>::size_type i = 0;
+    for (const auto& point : point_data) {
+      *iter_x = point.x();
+      *iter_y = point.y();
+      *iter_z = point.z();
+      *iter_intensity = intensity_data.at(i);
+
+      ++iter_x;
+      ++iter_y;
+      ++iter_z;
+      ++i;
+    }
+  }
+
+  // Error
+  else {
+    throw std::invalid_argument("Number of fields error");
+  }
+  return message;
+}
+
+TEST(TestPointCloud, XYZPointsUnorderedPC) {
+  const auto origin = Sophus::SE3d{};
+  // Define pointcloud params
+  std::vector<float>::size_type fields = 3;
+  unsigned int width = 1;
+  unsigned int height = 5;
+  // Create some raw data for the points
+  // clang-format off
+  const std::vector<Eigen::Vector3f> point_data = {
+    Eigen::Vector3f(1.0F, 2.0F, 3.0F),
+    Eigen::Vector3f(4.0F, 5.0F, 6.0F),
+    Eigen::Vector3f(7.0F, 8.0F, 9.0F),
+    Eigen::Vector3f(10.0F, 11.0F, 12.0F),
+    Eigen::Vector3f(13.0F, 14.0F, 15.0F)
+  };
+  // clang-format on
+  // Create point cloud message
+  const auto message = make_pointcloud<float, beluga_ros::msg::PointFieldF32>(fields, width, height, point_data);
+
+  // Check aligned pointcloud
+  auto cloud = beluga_ros::PointCloud3<beluga_ros::msg::PointFieldF32>(message, origin);
+  auto map = cloud.points();
+  // Check assert
+  for (unsigned i = 0; i < map.cols(); ++i) {
+    ASSERT_EQ(point_data.at(i).x(), map(0, i));
+    ASSERT_EQ(point_data.at(i).y(), map(1, i));
+    ASSERT_EQ(point_data.at(i).z(), map(2, i));
+  }
+
+  // Check sparse pointcloud
+  auto cloud_sparse = beluga_ros::PointCloudSparse3<beluga_ros::msg::PointFieldF32>(message, origin);
+  auto vector = cloud_sparse.points();
+  // Check assert
+  for (unsigned i = 0; i < vector.size(); ++i) {
+    ASSERT_EQ(point_data.at(i).x(), vector.at(i).x());
+    ASSERT_EQ(point_data.at(i).y(), vector.at(i).y());
+    ASSERT_EQ(point_data.at(i).z(), vector.at(i).z());
+  }
+}
+
+TEST(TestPointCloud, XYZPointsOrderedPC) {
+  const auto origin = Sophus::SE3d{};
+  // Define pointcloud params
+  std::vector<float>::size_type fields = 3;
+  unsigned int width = 3;
+  unsigned int height = 3;
+  // Create some raw data for the points
+  // clang-format off
+  const std::vector<Eigen::Vector3f> point_data = {
+    Eigen::Vector3f(1.0F, 2.0F, 3.0F),
+    Eigen::Vector3f(4.0F, 5.0F, 6.0F),
+    Eigen::Vector3f(7.0F, 8.0F, 9.0F),
+    Eigen::Vector3f(10.0F, 11.0F, 12.0F),
+    Eigen::Vector3f(13.0F, 14.0F, 15.0F),
+    Eigen::Vector3f(16.0F, 17.0F, 18.0F),
+    Eigen::Vector3f(19.0F, 20.0F, 21.0F),
+    Eigen::Vector3f(22.0F, 23.0F, 24.0F),
+    Eigen::Vector3f(25.0F, 26.0F, 27.0F)
+  };
+  // clang-format on
+  // Create point cloud message
+  const auto message = make_pointcloud<float, beluga_ros::msg::PointFieldF32>(fields, width, height, point_data);
+
+  // Check aligned pointcloud
+  auto cloud = beluga_ros::PointCloud3<beluga_ros::msg::PointFieldF32>(message, origin);
+  auto map = cloud.points();
```

Comment:
@pvela2017 nit: can be `const` I think. Same elsewhere.

Reply:
done 2c5197033c2b4b221f62ec3a51419db9bfb3dae0